charts/redpanda/values.yaml

```diff
@@ -252,6 +271,120 @@ storage:
+  # You can specify the amount of local storage you want to retain in local storage.
+  # You don't need to specify which log segments you want to move because Redpanda moves them automatically
+  # based on cluster-level configuration properties. Redpanda Tiered Storage indexes where data is offloaded,
+  # so it can retrieve the data when you need it.
+  #
+  # REF: https://docs.redpanda.com/docs/manage/tiered-storage/#tiered-storage-configuration-properties
+  #
+  # Topic configuration flags override any cluster-level configuration. For example, on new topics,
+  # if `cloud_storage_enable_remote_write` is set to `true`, you can set `redpanda.remote.write` to `false`
+  # to turn it off for a particular topic. Or, if you have Tiered Storage turned off for the cluster, you can enable it
+  # for individual topics with the topic flags. When cluster-level properties are changed,
+  # the changes apply only to new topics, not existing topics.
+  #
+  # REF: https://docs.redpanda.com/docs/manage/tiered-storage/#enable-tiered-storage-for-a-topic
+  tieredConfig:
+    # Global flag that enables Tiered Storage feature if license is provided.
+    # Default is false.
+    cloud_storage_enabled: false
+    # Cluster wide config option to enable data upload from Redpanda to cloud storage if set to true.
+    cloud_storage_enable_remote_write: true
+    # Cluster wide config option to enable Redpanda to fetch data from cloud storage if set to true.
+    cloud_storage_enable_remote_read: true
+    # This configuration option is available starting from 22.3.X
+    #
+    # Possible values:
+    # `config` - If IAM roles are not available, specify credentials in the cluster configuration file.
+    # `aws_instance_metadata` - For an AWS EC2 instance, use the instance metadata API from AWS.
+    # `sts` - For AWS on Kubernetes, use the Secure Token Service (STS).
+    # `gcp_instance_metadata` - For a VM running on GCP, or for Google Kubernetes Engine (GKE), use the instance metadata API from GCP.
+    #
+    # REF: https://docs.redpanda.com/docs/manage/security/iam-roles/#configuring-iam-roles
+    cloud_storage_credentials_source: config
+    # Cloud storage region. Required.
+    cloud_storage_region: ""
+    # Cloud storage bucket name. Required.
+    cloud_storage_bucket: ""
+    # Cloud storage access key used to authenticated if `cloud_storage_credentials_source` is set to `config`
+    # cloud_storage_access_key: ""
+    # Cloud storage secret key used to authenticated if `cloud_storage_credentials_source` is set to `config`
+    # cloud_storage_secret_key: ""
+    # API endpoint.
+    #
+    # - For AWS, this can be left blank. It’s generated automatically using the region and bucket.
+    #
+    # - For Google Cloud Service, use storage.googleapis.com.
+    cloud_storage_api_endpoint: ""
+
+    # Maximum size of the disk cache used by Tiered Storage.
+    # Default is 20 GiB.
+    cloud_storage_cache_size: 20GiB
```

Comment:
Does helm have knowledge of the size of drives?  If so, it should do something dynamic here, ideally coordinating the partition-per-shard limit with the disk size.

If not, then helm should not specify a value here, so that we can evolve the default in core.

Reply:
> Does helm have knowledge of the size of drives? 

Yes, it does. This exact property will drive the PersistentVolumeClaim which has size included. 
https://github.com/redpanda-data/helm-charts/blob/c4bbf8a34910bf4b0c984340a3688405024afb6d/charts/redpanda/templates/statefulset.yaml#L464

> If so, it should do something dynamic here, ideally coordinating the partition-per-shard limit with the disk size.

I'm happy to implement some logic here to drive partition-per-shard limit. Could you share more concrete example or algorithm that calculates it? 

> At some point we may change to not having an explicit cap by default, and if helm systems have all got an explicit 20GiB they'll get stuck that way.

If that happen we will update `values.yaml` accordingly to the new requirements. The volume size will always be necessary if user would like to configure persistent volumes. It can be driven from another place like Redpanda data volume size is defined https://github.com/redpanda-data/helm-charts/blob/c4bbf8a34910bf4b0c984340a3688405024afb6d/charts/redpanda/values.yaml#L262

Reply:
In most ideal situations we can give users an idea of how to calculate that based on emperical data (metrics, previous experience with customers etc...) If we have some input today, this is the best time to go write it down and provide an way to calculate but give users the option to override (by making this an override) our opinionated solution.

Reply:
> I'm happy to implement some logic here to drive partition-per-shard limit. Could you share more concrete example or algorithm that calculates it?

Probably makes sense to make it work the same way that the cloud team does, although they only have to worry about volume sizes of particular instance types.  For arbitrary volume sizes, you'd need to tune the partition limit, segment size + cache size together.  This is a bit too subtle for a PR comment thread: let's have a chat about it offline.

Reply:
> > I'm happy to implement some logic here to drive partition-per-shard limit. Could you share more concrete example or algorithm that calculates it?
> 
> Probably makes sense to make it work the same way that the cloud team does, although they only have to worry about volume sizes of particular instance types. For arbitrary volume sizes, you'd need to tune the partition limit, segment size + cache size together. This is a bit too subtle for a PR comment thread: let's have a chat about it offline.

Is that a prerequisite for merging this PR, or just a general principal we should work toward?

Reply:
>Is that a prerequisite for merging this PR, or just a general principal we should work toward?

I'm late to the party (the PR already merged), but yes, I think this should have been a blocking issue.

As soon as this is merged + visible to users, we will start putting clusters in the field with an explicit 20GiB limit, which can be too big for the disk, and can prevent Redpanda from migrating smoothly in future as/when we change the default to something smarter than a fixed number.